crates/astria-sequencer/src/sequence/action.rs

```diff
@@ -50,4 +73,58 @@ impl Action {
 }
 
 #[async_trait::async_trait]
-impl ActionHandler for Action {}
+impl ActionHandler for Action {
+    async fn check_stateful<S: StateReadExt + 'static>(
+        &self,
+        state: &S,
+        from: &Address,
+    ) -> Result<()> {
+        let curr_balance = state
+            .get_account_balance(from)
+            .await
+            .context("failed getting `from` account balance")?;
+        let fee = SEQUENCE_ACTION_FEE_PER_BYTE * self.data.len() as u128;
```

Comment:
I think a `fn calculate_sequencer_fee<T: AsRef<[u8]>>(data: T)` (or just `data: &[u8]`) + unit tests would be great for this.